<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.269">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Self-Guided Learning - Learning Resources</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/cookie-consent/cookie-consent.js"></script>
<link href="site_libs/cookie-consent/cookie-consent.css" rel="stylesheet">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>

<script type="text/javascript" charset="UTF-8">
document.addEventListener('DOMContentLoaded', function () {
cookieconsent.run({
  "notice_banner_type":"simple",
  "consent_type":"implied",
  "palette":"light",
  "language":"en",
  "page_load_consent_levels":["strictly-necessary","functionality","tracking","targeting"],
  "notice_banner_reject_button_hide":false,
  "preferences_center_close_button_hide":false,
  "website_name":""
  });
});
</script> 
  
<style>html{ scroll-behavior: smooth; }</style>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Self-Guided Learning</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./software-installs.html">
 <span class="menu-text">Software Installs</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./learning-resources.html" aria-current="page">
 <span class="menu-text">Learning Resources</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./domain-specific-series.html">
 <span class="menu-text">Domain-Specific Series</span></a>
  </li>  
</ul>
              <div class="quarto-toggle-container">
                  <a href="" class="quarto-reader-toggle nav-link" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
              </div>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#languages-and-programming" id="toc-languages-and-programming" class="nav-link active" data-scroll-target="#languages-and-programming">Languages and Programming</a></li>
  <li><a href="#plotting-and-visualizations" id="toc-plotting-and-visualizations" class="nav-link" data-scroll-target="#plotting-and-visualizations">Plotting and Visualizations</a></li>
  <li><a href="#single-cell-analyses" id="toc-single-cell-analyses" class="nav-link" data-scroll-target="#single-cell-analyses">Single-Cell Analyses</a></li>
  <li><a href="#proteomics" id="toc-proteomics" class="nav-link" data-scroll-target="#proteomics">Proteomics</a></li>
  <li><a href="#imaging-analysis" id="toc-imaging-analysis" class="nav-link" data-scroll-target="#imaging-analysis">Imaging Analysis</a></li>
  <li><a href="#workflow-managers" id="toc-workflow-managers" class="nav-link" data-scroll-target="#workflow-managers">Workflow Managers</a></li>
  <li><a href="#open-science-reproducibility" id="toc-open-science-reproducibility" class="nav-link" data-scroll-target="#open-science-reproducibility">Open Science &amp; Reproducibility</a></li>
  <li><a href="#cloud-computing" id="toc-cloud-computing" class="nav-link" data-scroll-target="#cloud-computing">Cloud Computing</a></li>
  <li><a href="#mlai" id="toc-mlai" class="nav-link" data-scroll-target="#mlai">ML/AI</a></li>
  <li><a href="#general-bioinformatics" id="toc-general-bioinformatics" class="nav-link" data-scroll-target="#general-bioinformatics">General Bioinformatics</a></li>
  <li><a href="#statistics" id="toc-statistics" class="nav-link" data-scroll-target="#statistics">Statistics</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content column-page-left" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Learning Resources</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>A collection of self-guided learning resources created by others.</p>
<section id="languages-and-programming" class="level3">
<h3 class="anchored" data-anchor-id="languages-and-programming">Languages and Programming</h3>
<details>
<summary>
<strong>Python</strong>
</summary>
<p><a href="https://www.youtube.com/watch?v=mvK0UzFNw1Q">How to Learn to Code</a> – A 5 minute video on how to approach programming, either learning theory, soliciting advice, OR (hint hint) <strong><em>problem solving</em></strong>.</p>
<p><a href="https://online.umich.edu/courses/programming-for-everybody-getting-started-with-python/">Python Course: Programming for Everybody (Getting Started with Python)</a> – A terrific free course by the University of Michigan to learn the basics of programming in Python, with no prerequisites.</p>
<p><a href="https://www.youtube.com/watch?v=ocA2IMe7dpA&amp;list=PLhYDP66xNTKRdtUM8ekEmq1fC0mosXrgn&amp;index=15">Biopython Course: Bioinformatics with Biopython</a> – A 1-hour YouTube course that covers some common uses of biopython. Biopython is a set of tools for biological computaiton that is written in Python. It has many useful functions for processing and working with sequence files.</p>
<p><a href="https://www.pythoncheatsheet.org/">Python Cheatsheet</a> – A quick-reference, cheatsheet-styled website for common tasks/tricks in python. Helpful for beginners and power-users.</p>
</details>
<details>
<summary>
<strong>R</strong>
</summary>
<p><a href="https://swirlstats.com/">R Course: swirl - learn R, in R</a> – Super cool way to learn R programming and datascience interactively, at your own pace, and in the R console. There is a library of courses you can learn from.</p>
<p><a href="https://www.youtube.com/playlist?list=PLhR2Go-lh6X4fCAa3c_TBAjZD5A25s7fo">R Course: Data Analysis using R</a> – A comprehensive and high quality YouTube based course by Dr.&nbsp;Danny Arends (with videos, access to lectures, assignments, and answers) for those within minimal to no previous programming experience. This might be one of the best courses available.</p>
<p><a href="https://r4ds.had.co.nz/index.html">R for Data Science</a> – A website for the “R for Data Science” book, with a focus on how to perform data science with R, from data exploration (visualization, workflow, working with tidy data, scripting), wrangling, modeling, and communicating results (R markdown).</p>
<p><a href="https://cran.r-project.org/doc/contrib/Baggott-refcard-v2.pdf">R Cheatsheet</a> – A quick-reference cheatsheet for R.</p>
</details>
<details>
<summary>
<strong>Unix and Bash</strong>
</summary>
<p><a href="https://seankross.com/the-unix-workbench/introduction.html">Unix Book: Unix Workbench</a> – A website (book) for those new to Unix-like operating systems and working at the command-line. This book covers unix and command-line basics, as well as introductory bash programming concepts (math, variables, loops, input/output, arrays, pbraces, functions), and writing programs. As a nice bonus, it also gives brief introductions to Git, GitHub, and Cloud Computing.</p>
<p><a href="https://github.com/Idnan/bash-guide">Bash Reference: bash-guide</a> – Common commands, with some minimal examples.</p>
</details>
<details>
<summary>
<strong>Julia</strong>
</summary>
<p><a href="https://github.com/Datseris/Zero2Hero-JuliaWorkshop">Julia Course: Zero-to-Hero Julia Workshop</a> – Two day intensive, recorded (<a href="https://www.youtube.com/watch?v=Fi7Pf2NveH0">YouTube</a>), workshop on the programming language, Julia. This workshop is intended for beginners. Julia is a programming language designed for scientific computing, to be quick, but also readable and high-level like Python.</p>
</details>
</section>
<section id="plotting-and-visualizations" class="level3">
<h3 class="anchored" data-anchor-id="plotting-and-visualizations">Plotting and Visualizations</h3>
<details>
<summary>
<strong>With Python</strong>
</summary>
<p><a href="https://gilberttanner.com/blog/introduction-to-data-visualization-inpython/">Introduction to Data Visualization in Python</a> – A very simple introduction to data visualization in Python.</p>
<p><a href="https://realpython.com/tutorials/data-viz/">Python Data Visualization Tutorials</a> – A set of tutorials for plotting with Python in Pandas, matplotlib, seaborn, and boken (interactive visualizations).</p>
<p><a href="https://www.kaggle.com/code/kanncaa1/plotly-tutorial-for-beginners">Interactive Visualizations with Plotly in Python</a> – An introduction, for beginners, to interactively plotting in Python using Plotly.</p>
</details>
<details>
<summary>
<strong>With R</strong>
</summary>
<p><a href="https://rkabacoff.github.io/datavis/">Data Visualization with R</a> – A Web-based book focused on data visualization and plotting with R. It covers data importing, cleaning, using plotting libraries (ggplot2), customizing plots, interactive visualization, and more.</p>
<p><a href="https://shiny.rstudio.com/images/shiny-cheatsheet.pdf">Shiny Cheatsheet</a> – A quick reference cheatsheet for creating interactive web apps with shiny.</p>
</details>
</section>
<section id="single-cell-analyses" class="level3">
<h3 class="anchored" data-anchor-id="single-cell-analyses">Single-Cell Analyses</h3>
<details>
<summary>
<strong>Single-Cell RNA-seq</strong>
</summary>
<p><a href="https://github.com/cellgeni/scRNA.seq.course">scRNA-seq Analysis Tutorial: Analysis of Single-Cell RNA-seq Data</a> – A continuously updated and comprehensive single-cell RNA-seq analysis tutorial/course, <strong>taught primarily in <code>R</code></strong>. This course begins with a discussion of single-cell methods, experimental design, and data processing and ends with single-cell dataset integration. It uses primarily <code>Seurat</code>, but also covers other tools for analysis and integration.</p>
<p><a href="https://www.sc-best-practices.org/preamble.html">scRNA-seq Tutorial: Single-cell best practices</a> – A very good tutorial on the best practices in single-cell RNA-seq analysis, <strong>taught primarily in <code>Python</code></strong>. This course starts with pre-processing and QC, and ends with a brief overview on CITE-seq, immune repertoire, and integration. The <a href="https://www.embopress.org/doi/full/10.15252/msb.20188746">paper</a> that this course was originally based on.</p>
<p><a href="https://satijalab.org/scgd22/">Single-Cell Genomics Workshop: Single-Cell Genomics Day</a> – A yearly workshop by the Satija lab (leaders in the world of single-cell analysis) on various single-cell genomics analysis topics/methods. In addition to a “recent and future advances” session, the workshop covers spatial and temporal analysis, epigenomic analysis, genotype-phenotype landscapes, and multimodal analyses.</p>
<p><a href="https://nf-co.re/scrnaseq">scRNA-seq Best Practices Pipeline: nf-core/scrnaseq</a> – A best-practices pipline for processing 10x genomics single-cell data using <code>Nextflow</code>, a workflow management tool that provides improved computational metics and reproducibility.</p>
<p><a href="https://www.nxn.se/single-cell-studies/">Curated database of single cell studies</a> – A manually curated database of over 1800 single-cell studies, dating back to 2002. The doi, number of cells, organism, tissue, and experimental method are included, as well as other useful information.</p>
</details>
<details>
<summary>
<strong>Single-Cell ATAC-seq</strong>
</summary>
<p><a href="https://liulab-dfci.github.io/bioinfo-combio/scatac.html">scATAC-seq Lectues: Best practices for ATAC-seq</a> – A set of lectures from Ming Tang that cover the scATAC-seq experimental method, pre-processing and QC, and anlysis and integration.</p>
<p><a href="https://www.sciencedirect.com/science/article/pii/S2001037020303019#f0005">scATAC-seq Review Paper: Single-cell ATAC sequencing analysis: From data preprocessing to hypothesis generation</a> – A helpful review of scATAC-seq technologies and analysis software.</p>
</details>
</section>
<section id="proteomics" class="level3">
<h3 class="anchored" data-anchor-id="proteomics">Proteomics</h3>
<p>[Mass-Spec Based Proteomics Aanalysis Tutorial]</p>
<p><a href="https://www.nature.com/articles/s41596-021-00566-6">Best practices for mass-spec-based biomarker discovery</a> – A high-level paper overview of best practices for mass-spectrometry-based biomarker discovery.</p>
</section>
<section id="imaging-analysis" class="level3">
<h3 class="anchored" data-anchor-id="imaging-analysis">Imaging Analysis</h3>
<p>[Segmentation]</p>
</section>
<section id="workflow-managers" class="level3">
<h3 class="anchored" data-anchor-id="workflow-managers">Workflow Managers</h3>
<p>[Nexflow]</p>
<p>[Snakemake]</p>
<p>[Galaxy]</p>
</section>
<section id="open-science-reproducibility" class="level3">
<h3 class="anchored" data-anchor-id="open-science-reproducibility">Open Science &amp; Reproducibility</h3>
<p>[Posit]</p>
<p><a href="https://posit.co/resources/cheatsheets/?type=posit-cheatsheets/">Posit Cheatsheets</a> – An amazing list of cheatsheets for eveyrthing rmarkdown, shiny, reticulate, stringr, readr, sparkly, tidyr, dplyr, ggplot2, and more.</p>
<p>[Jupyterlab and Notebooks]</p>
</section>
<section id="cloud-computing" class="level3">
<h3 class="anchored" data-anchor-id="cloud-computing">Cloud Computing</h3>
<p>[]<br>
[]<br>
[]</p>
</section>
<section id="mlai" class="level3">
<h3 class="anchored" data-anchor-id="mlai">ML/AI</h3>
</section>
<section id="general-bioinformatics" class="level3">
<h3 class="anchored" data-anchor-id="general-bioinformatics">General Bioinformatics</h3>
<p><a href="https://www.youtube.com/playlist?list=PLhR2Go-lh6X5A5WbiO3SPHuoWbwpNznUl">Bioinformatics Course: Bioinformatics for Plant and Animal Sciences</a> – A quality YouTube based course by Dr.&nbsp;Danny Arends (with <a href="https://www.youtube.com/playlist?list=PLhR2Go-lh6X5A5WbiO3SPHuoWbwpNznUl">videos</a>, access to <a href="https://dannyarends.nl/bioinformatik/">lectures</a>, <a href="https://dannyarends.nl/bioinfo/">assignments</a>, and answers) on bioinformatics. This course covers biological topics (genetics, molecular biology, metabolism, homology/phylogeny, etc.), as well as computational and analysis topics (transcriptomics, R, stats) and ends with learning how to create an R package.</p>
<p><a href="https://rosalind.info/problems/locations/">Rosalind</a> – a great resource to learn about bioinformatics (including algorithms) and programming by problem-solving.</p>
</section>
<section id="statistics" class="level3">
<h3 class="anchored" data-anchor-id="statistics">Statistics</h3>
<p><a href="https://github.com/gladstone-institutes/Bioinformatics-Workshops/wiki/Introduction-to-Experimental-Design-and-Hypothesis-testing">Introductory Stats Course: Introduction to Experimental Design and Hypothesis Testing</a> – A beginner-friendly set of resources (slides, code) from an online workshop by the Gladstone Institutes to work though. The resources cover concepts underlying hypothesis testing.</p>
<p><a href="https://github.com/gladstone-institutes/Bioinformatics-Workshops/wiki/Statistics-of-Enrichment-Analyses-Methods">Intermediate Course: Statistics of Enrichment Analyses Methods</a> – A course designed for those familiar with basic statistics and experimental design concepts, an understanding of high-throughput analyses (RNA-seq, Mass Spec, etc.), and a working knowledge of R.</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
      <div class="nav-footer-center"><div class="cookie-consent-footer"><a href="#" id="open_preferences_center">Cookie Preferences</a></div></div>
  </div>
</footer>



<script src="site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>